src/api/apiHelpers.ts

```diff
@@ -139,15 +139,17 @@ export async function handleGeneratedApiResponse(response: Response): Promise<an
     return {}
   }
 
+  const responseBody = await response.json()
+
   if (!response.ok) {
     throw new ApiError(
       response.status,
-      response.body,
+      responseBody,
       `Échec de la requête ${response.url}, code: ${response.status}`
     )
   }
 
-  return await response.json()
+  return responseBody
 }
```

Comment:
je pense ce serait plus safe avec un truc du genre: https://stackoverflow.com/questions/37121301/how-to-check-if-the-response-of-a-fetch-is-a-json-object-in-javascript

```tsx
// en dehors
const extractResponseBody = async (response: Response): Promise<string> => {
  const contentType = response.headers.get('content-type')
  if (contentType && contentType.indexOf('application/json') !== -1) {
    return await response.json()
  }
  return await response.text()
}

  const responseBody = extractResponseBody(response)

  if (!response.ok) {
    throw new ApiError(
      response.status,
      responseBody,
      `Échec de la requête ${response.url}, code: ${response.status}`
    )
  }

  return responseBody
```